modules/cache/main.tf

```
@@ -11,7 +12,14 @@ locals {
     var.tags,
   )
 
-  cache_bucket_name = var.cache_bucket_name_include_account_id ? "${var.cache_bucket_prefix}${data.aws_caller_identity.current.account_id}-gitlab-runner-cache" : "${var.cache_bucket_prefix}-gitlab-runner-cache"
+  cache_bucket_string = var.cache_bucket_name_include_account_id ? format("%s%s-gitlab-runner-cache", var.cache_bucket_prefix, data.aws_caller_identity.current.account_id) : format("%s-gitlab-runner-cache", var.cache_bucket_prefix)
+  cache_bucket_name   = var.cache_bucket_set_suffix ? format("%s-%s", local.cache_bucket_string, random_string.s3_suffix.result) : local.cache_bucket_string
+}
+
+resource "random_string" "s3_suffix" {
+  length  = 8
```

Please can you add here a count so no changes are suggested if suffix is not used
```suggestion
  count = var.cache_bucket_set_suffix ? 1 : 0 
  length  = 8
```